tests/foreman/api/test_product.py

```diff
@@ -107,6 +107,33 @@ def test_positive_update_1(self, attrs):
         self.product = self.product.update()
         self.assertEqual(getattr(self.product, attrs[0]), attrs[1])
 
+    @data(
+        (u'name', gen_string('alphanumeric', randint(1, 255))),
+        (u'name', gen_string('alpha', randint(1, 255))),
+        (u'name', gen_string('cjk', randint(1, 85))),
+        (u'name', gen_string('latin1', randint(1, 255))),
+        (u'name', gen_string('numeric', randint(1, 255))),
+        (u'name', gen_string('utf8', randint(1, 85))),
+    )
+    def test_positive_update_2(self, attrs):
```

Comment:
In no particular order:

* `attrs` is an unused variable.
* `attrs` is a single-item dict. It's simpler to pass in just a name, rather than a dict containing a name.
* The use of `getattr` and `setattr` is unnecessary in this context.

Try something like this instead:

```python
    @data(
        gen_string('alphanumeric', randint(1, 255)),
        gen_string('alpha', randint(1, 255)),
        gen_string('cjk', randint(1, 85)),
        gen_string('latin1', randint(1, 255)),
        gen_string('numeric', randint(1, 255)),
        gen_string('utf8', randint(1, 85)),
    )
    def test_positive_update_2(self, new_name):
        """@Test: Rename Product back to original name

        @Feature: Product

        @Assert: Product Renamed to original

        """
        old_name = self.product.name
        self.product.name = new_name
        self.assertEqual(self.product.name, self.product.update(['name']).name)
        self.product.name = old_name
        self.assertEqual(self.product.name, self.product.update(['name']).name)
```